### ISSUE TITLE
Recolor causes crash on pre Lollipop

### ISSUE THREAD

**Issue by twyatt** [andkulikov/Transitions-Everywhere#45]
Tested the sample app on 4.1.1 and pressing the "TAP ON ME" button in the "Recolor transition" sample causes a StackOverflowError:

```java
com.andkulikov.transitionseverywhere E/AndroidRuntime: FATAL EXCEPTION: main
  java.lang.StackOverflowError
      at java.lang.Integer.<init>(Integer.java:88)
      at java.lang.Integer.valueOf(Integer.java:707)
      at android.util.IntProperty.set(IntProperty.java:45)
      at android.util.IntProperty.set(IntProperty.java:45)
      at android.util.IntProperty.set(IntProperty.java:45)
      at android.util.IntProperty.set(IntProperty.java:45)
      ...
```

On Android 5.0.0 the same sample works as expected.

Replacing `COLORDRAWABLE_COLOR ` with `"color"` at [Recolor.java#L124](https://github.com/andkulikov/Transitions-Everywhere/blob/master/library/src/main/java/com/transitionseverywhere/Recolor.java#L124) fixes the issue:
```java
    return ObjectAnimator.ofObject(endColor, "color",
            new ArgbEvaluator(), startColor.getColor(), finalColor);
```

**Comment by andkulikov** [comment]
Hm. Yeah. its because of a bug fixed only in 4.2.2. 
https://android.googlesource.com/platform/frameworks/base/+/c5d43f76fd7c3ccb91f1b75618a9c9e8f202505b
I didn't know about it

**Comment by twyatt** [comment]
Eek, I had no idea either. I just took to easiest (admittedly also quite lazy) approach to preventing the Stackoverflow with the caveat that providing `ObjectAnimator#ofObject` with the `"color"` parameter means it will use Reflection to call the `setColor` method.

Alternatively a `ValueAnimator` could be used to remove the use of Reflection?

**Comment by andkulikov** [comment]
Basically we can use just Property, not IntProperty. There is will be no reflection, but autoboxing for Integer->int. 
But I want to use IntProperty starting for 4.2 in that case. I need to think about it more.

**Comment by andkulikov** [comment]
fixed in 1.6.8. thank you!
